### DISCUSSION TITLE
How to put Filament assets together with Vite assets?

### DISCUSSION THREAD

**Discussion by man0l** [filamentphp/filament#9029] [Help] [answered]
Hello, I am new to fillament. I will appreciate your help. I want to achieve a single command build, because I have both php artisan filament:assets and vite.
I am with clean installation of Laravel and Filament and I am settuping an application.

When I publish the assets by running  php artisan filament:assets they go the public/ directory.
I also have assets defined in vite.config.js from the resources/ directory. 

In architecrtural perspective, how can I combine the assets and run only one command for the build (I need dev and prod build)? I am not  sure if this is the correct approach, so I would ask for more advanced aproach if there is any.

Sincerely,
Manol.

**Comment by leandrocfe** [comment]
Not sure if I got the idea. Could you share the repo on GitHub?

**Reply by man0l** [reply]
Unforutanely it is a company project and I cannot share it.
My idea is to build all assets from a single place - I preffer using vite for building the assets. I want to ditch the command php artisan filament:assets and configure vite, something like this:

![image](https://github.com/filamentphp/filament/assets/1413216/bbe99d79-e992-449b-801c-d20abdd22078)

The approach from the screenshot has it's negative sides because it has issues with the external libraries - in the example, it cannot build echo.js and I am not sure why.

I am not sure if this is the correct approach for building with vite. Would you advise me?

**Reply by leandrocfe** [reply, marked as answer]
What about [this](https://filamentphp.com/docs/3.x/support/assets#registering-javascript-files) ? When you run filament:assets command this is copied into the /public directory.

**Reply by man0l** [reply]
Thank you, I've been using it

**Comment by nagi1** [comment]
Hey @man0l I'm interested on building filament assets using Filament have you got this working?

**Comment by man0l** [comment]
Hi @nagi1  this is my current setup:

![image](https://github.com/user-attachments/assets/67e09f5f-54f4-4e69-b882-6980abf392eb)

I've used vite to build all the assets, here's my pacpkage.json

![image](https://github.com/user-attachments/assets/d1db019e-5e5e-4f82-a435-b6adce069559)

I build the assets as well during build:

            "@php artisan filament:assets --ansi --no-interaction",

This is the end result:

![image](https://github.com/user-attachments/assets/e3081fb5-6291-4df4-a608-49d92effc237)
![image](https://github.com/user-attachments/assets/d43a9a7b-36fa-4e55-9215-47d3f928534b)

e.g. I didn't manage to combine it but it works
